packages/std/src/errors/system_error.rs

```diff
@@ -16,9 +16,10 @@ use crate::HumanAddr;
 #[serde(rename_all = "snake_case")]
 #[non_exhaustive]
 pub enum SystemError {
-    InvalidRequest { error: String },
+    InvalidRequest { error: String, request: Vec<u8> },
+    InvalidResponse { error: String, response: Vec<u8> },
```

Comment:
Can we change the `request`/`response` to `Binary`, which gives us a proper base64 encoded JSON representation?

Reply:
Changing it to binary will mean that the printed text will be in Hex, rather than lossy utf8 (see just a few lines below). so for example if the error is just due to an incorrectly structured json, you'll later see `7B226669656C64223A2276616C7565222C...` instead of `{"field":"value",...`.
On the other hand if it contains invalid utf8 you'll see a lot of ���

Reply:
I think you are confusing different layers here. When serializing SystemError in JSON (invisible must of the time), `request: Vec<u8>` becomes something like `[0, 123, 43, 87, 2, 65, ...]`. When using `Binary` it is base64 encoded data. The error display implementation then becomes:

```Rust
            SystemError::InvalidResponse { error, response } => write!(
                f,
                "Cannot parse response: {} in: {}",
                error,
                String::from_utf8_lossy(response.as_slice())
            ),
```

Reply:
ahhh, i see. yeah that makes sense, i'll do that

Reply:
done

Reply:
done